pkg/api/types.go

```diff
@@ -1189,6 +1189,70 @@ type PreferredSchedulingTerm struct {
 	Preference NodeSelectorTerm `json:"preference"`
 }
 
+// The node this Taint is attached to has the effect "effect" on
+// any pod that that does not tolerate the Taint.
+type Taint struct {
+	// Required. The taint key to be applied to a node .
+	Key string `json:"key" patchStrategy:"merge" patchMergeKey:"key"`
+	// Required. Value is the taint value the toleration matches to.
+	Value string `json:"value,omitempty"`
+	//Required. The effect of the taint, ref TaintEffect.
```

Comment:
// Required. The effect of the taint on pods that do not tolerate the taint. Valid effects are NoSchedule and PreferNoSchedule.